test/src/unit-sparse-global-order-reader.cc

```diff
@@ -958,22 +960,39 @@ static std::optional<bool> can_complete_in_memory_budget(
     using BitmapType = uint8_t;
 
     size_t data_size = 0;
+    size_t bitmap_size = 0;
+
     for (size_t d = 0;
          d < std::tuple_size<decltype(instance.dimensions())>::value;
          d++) {
       data_size +=
           fragment_metadata[f]->tile_size("d" + std::to_string(d + 1), t);
     }
 
+    if (!instance.subarray.empty()) {
+      bitmap_size += fragment_metadata[f]->cell_num(t) * sizeof(BitmapType);
+    }
+
+    if (instance.condition.has_value()) {
+      // query condition fields are eagerly materialized
+      std::unordered_set<std::string> fields;
+      instance.condition.value()->get_field_names(fields);
+
+      for (const auto& field : fields) {
+        if (!array_schema.is_dim(field)) {
+          data_size += tiledb::sm::ReaderBase::get_attribute_tile_size(
+              array_schema, fragment_metadata, field, f, t);
+        }
+      }
+
+      bitmap_size += fragment_metadata[f]->cell_num(t) * sizeof(BitmapType);
```

Comment:
If we have both a `subarray` and a query condition we need double the `bitmap_size` ? Seems odd, isn't the `bitmap` just one?

Reply:
There are actually two bitmaps!  One of the bitmaps indicates whether a cell passes the subarray, and the other seems to be used for everything else.

Coincidentally enough, the proptest found an example of why we want this in the failing CI (and I expect resolution of this to be test-only changes).

If you have a query condition `attribute = 0` and fragment 0 has `attribute = 0` and fragment 2 has `attribute = 1` for the same coordinate, then you expect to not see that cell in the result set.

So, we use the first bitmap to evaluate the subarray and determine whether to put coordinates in the de-duplication queue.  And then we use the second bitmap after pulling cells out of the de-duplication queue to determine whether they should be emitted.

Reply:
Thank you for the clarification!